### ISSUE TITLE
Step 5 of Procurement Tracker - Pre-award (Submitter/Requester)

### ISSUE THREAD

**Issue by kimschulke** [HHS/OPRE-OPS#2279]
### User Story
As a team member listed on the agreement, I need to request pre-award approval, so that my Division Director can review the final state of the contract before I send the Final Consensus Memo to the Procurement Shop

### Preconditions
* An agreement has one or more budget line in `Executing` status and the contract has already advanced past the 4th step of the procurement tracker.

### Acceptance Criteria
- [ ] The user can see a visual understanding of the overall procurement process and the current progress in it
- [ ] The user can optionally choose a target completion date and optionally commit the input of only that information and nothing else--independent of other user input.
- [ ] The user can request pre-award approval and review the agreement details before committing to move forward/send to approval
- [ ] Any users accessing the agreement after this state whilst in review should be aware that this agreement is in pre-award review.

- [ ] Only after all approvals have been received, the user can attest that final edits were made/approved and they received pre-award approval and sent the final consensus memo to the procurement shop
- [ ] The user can select the OPS user who attests to this and the date that this was completed. The date must either be the current or a prior date.
- [ ] The user can optionally provide any notes limited to 750 characters
- [ ] The user can reset all user input
- [ ] The user can commit their input which will advance the contract to to the next step
- [ ] If any error is encountered in persisting the change(s), that error should be presented to the user
- [ ] The relevant entries should be recorded in the event/history table(s) and be able to be properly rendered properly in the history of changes to the agreement

-----------------------------------------

#### UX
- [x] Explore design options
- [x] Share design with team to discuss development considerations
- [x] Refine the design based on feedback
- [x] Share designs with stakeholders and other OPRE team members
- [x] Refine the design based on feedback
- [x] Provide design assets with annotations to developers

### Definition of Done Checklist
- [ ] UI works as designed (UX team)
- [x] PR(s) have been merged to main
- [ ] Design/tech debt eliminated
- [ ] Build process updated
- [ ] Documentation updated or added
- [ ] Feature flags/toggles created

### Additional Context & Resources
* #1517, #1501, #967

**Comment by kimschulke** [comment]
Hey @josbell I am finally able to test this in staging and noticed something with the Documents Tab tooltip. Its displaying as a dark grey rectangle instead of a tooltip like it should. See below. Additionally, the document placeholder should have a rounded corner radius. Shall I move this back to in progress?

<img width="1273" height="870" alt="Image" src="https://github.com/user-attachments/assets/efe84075-5f70-4fff-8a0d-6279dc23be2f" />

**Comment by kimschulke** [comment]
I also get an unexpected behavior after clicking Send to Approval. The In Review alert displays, but its not triggered at the right time. Additionally, a random "changes approved" alert shows up instead of the success for sending to approval.

What should happen is - user clicks send to approval --> user sees success that request was sent to DD --> after some time the in review alert pops up.

Whenever the in review alert is displayed, that means the agreement cannot be edited. But so far, it looks like I can still edit the agreement despite the alert saying I cannot.

**UPDATE**: I think some of this will be resolved via #1639 and #5577 

<img width="1336" height="1216" alt="Image" src="https://github.com/user-attachments/assets/5d21b2e3-1c67-4187-bbbf-ea36bfa6f365" />

**Comment by kimschulke** [comment]
Here is a screen capture that shows the issues above, as well as captures the interactions between steps. Right now, when you complete a step, the screen scrolls you down to the bottom of the next step, but it should stay focused on the step completion so they can their progress was saved before moving ahead.

https://github.com/user-attachments/assets/c8115800-fdb3-493d-92c4-c212ab09c02b

**Comment by npage123** [comment]
After testing with @josbell @andressacurvelo on the DD leaving a note on the review agreement page before approving, this is what appears when logged in as user demo when navigating to the agreement.

<img width="1388" height="785" alt="Image" src="https://github.com/user-attachments/assets/1e4eb0c8-0e6d-432d-ad16-aab01460fd6c" />

**Comment by jonnalley** [comment]
@josbell please do review DoD when you can please

**Comment by kimschulke** [comment]
@josbell and @kimschulke collabed today on the step interactions. Josbell set it up so the step completion stays focused on the current/completed step so the user can see their action to complete the step was successful. This felt important in this use case because we do not have a success message to show them otherwise. The next step will still become active and the accordion will open, but it will not scroll down the next step. It will stay focused on the current/completed step and the user will scroll down to continue if they'd like to.

Our back-up option (back pocket, if needed) would be to disable the interaction to open the next step/accordion all together. This will ensure the user stays focused on the current/completed state as a successful completion, and then they can explore the next step at their own leisure. The step indicator would still show the active step, but the accordion would not be opened automatically. Upon exiting or re-navigating to the tracker, the active state would be open by default.

In the end, theres a trade-off: is it more important that the user feels confident that their action/completing a step was successful? or is it more important that they are moved ahead automatically to the next step w/o knowing if their prior action was completed? I personally think the 1st item is more important in this case and has a less detrimental impact on the user experience. Knowing something they did was successful feels more important that being shuffled to the next step, when it might also be pretty intuitive from a process-perspective that there is more to be done in the procurement tracker after they complete step 1 or step 2. They will know that inherently from a business-perspective.

**Comment by jonnalley** [comment]
@kimschulke so is there any action items left on this particular aspect of step 5 then in this story? I didn't read any particular action items that @josbell  didn't already fix unless I'm mistaken ?? please set me straight. and does this mean that a UX review has been conducted for DoD purposes ?

**Comment by kimschulke** [comment]
@jonnalley We still need to address the alerts, but maybe we can take care of that in https://github.com/HHS/OPRE-OPS/issues/1639

@josbell let me know what you think

**Comment by jonnalley** [comment]
moving to review column based on prior comments and progress

**Comment by josbell** [comment]
Latest updates overview
@npage123 @kimschulke @andressacurvelo 

Updated Procurement Tracker Step 5 description text

1. Changed "HHS Consolidated Acquisition Solution (HCAS)" to "Procurement Shop" to clarify the workflow (Pre-Award Approval goes to Procurement Shop first, not HCAS)
2. Fixed missing obligate-by date handling in Budget Team review cards
3. Added auto-dismiss timeout to pre-award approval notifications
4. Fixed BLI grouping when services_component_number is missing


**Test 1: Verify Step 5 description text updates**

1. Navigate to an agreement with procurement tracker (e.g., Agreement 13 or 14)
2. Complete steps 1-4 to reach Step 5
3. Expected: Step 5 description reads:
4. "All agreements need Pre-Award Approval before the Final Consensus Memo can be sent to the Procurement Shop."
5. Second paragraph mentions "Once you receive Pre-Award Approval, and the Budget Team submits the requisition, upload the Final Consensus Memo to the HHS Consolidated Acquisition Solution (HCAS)"
6. Expected: Text is split into two paragraphs with white space between them

**Test 2: Verify Budget Team review card layout handles missing dates**

1. Log in as Budget Team user
2. Navigate to Change Requests page (/change-requests)
3. Find a pre-award requisition review card
4. Expected: Card always shows consistent 5-column grid layout
5. Expected: "Obligate By Date" field shows date value OR "None" (never missing)
6. Expected: "Agreement Total" always appears in columns 5-7 (right side)

**Test 3: Verify auto-dismiss for pre-award approval notifications**

1. As a COR, request pre-award approval for an agreement
2. As a Division Director, approve the pre-award request
3. Return to agreement page as COR
4. Expected: Green success notification appears at top of page
5. Expected: After 6 seconds, notification automatically dismisses
6. Repeat steps 1-2, but this time DECLINE the request
7. Expected: Red error notification appears
8. Expected: After 6 seconds, declined notification also auto-dismisses (matches global Alert behavior)

**Test 4: Verify BLI grouping with missing service component numbers**

1. Navigate to agreement review-pre-award page with BLIs that have services_component_id but missing services_component_number (test data: Agreement 14)
2. Expected: BLIs are grouped correctly by service component number (looked up by ID)
3. Expected: BLIs are NOT incorrectly grouped under "BLs not associated with a Services Component"

**Comment by kimschulke** [comment]
remaining to dos as discussed in team collab today --

- [x] Add agreement completion validation during pre-award request (check if agreement has empty states and if so, show red errors notification with red error states on each component and show a button to edit and correct errors, then go back to pre-award request.) @josbell

**Comment by jonnalley** [comment]
@kimschulke and @josbell to recreate the scenario locally and simulate to prove its fixed.

**Comment by kimschulke** [comment]
@josbell and I paired today on this and noticed a few things to adjust in the current implementation

- [x] for the error state on budget lines, only underline column/data that is in an error state (not the whole row)
- [x] When they click save, if their change needs an approval, show the modal, if not just go back to pre-award
- [x] When they get back to pre-award, show the in review alert and disable the button to send to approval because they need to wait until their are no in review items pending (no other alerts needed)

[All of this is mocked up here](https://www.figma.com/design/dsQFe6G2KjcTWNq7NNZujm/OPRE-OPS?node-id=33991-66200&t=R44K6q74g5gkuCP4-1) (still refining a few minor details)

**Comment by kimschulke** [comment]
@josbell Looks awesome! Only a few more items:

- [x] Add tooltip on hover for any budget lines In Review Status (it explains whats in review)
- [x] Update text on tooltip for disabled Send to Approval button after they resolve error and return to the Pre-Award Request page


Thank you!

- [x] And for a separate GH issue - look into consistency on Amount component to ensure we display the $ sign

**Comment by npage123** [comment]
Reviewing with @josbell and @andressacurvelo , we decided the:

- [x] In Review BL tooltips should replace the words 'null' and 'undefined' with 'TBD'.

<img width="1124" height="616" alt="Image" src="https://github.com/user-attachments/assets/cae3940e-b344-4566-a09d-476100caa829" />

**Comment by npage123** [comment]
@josbell Ran into a bug on staging in which I declined the submission as a DD and then signed back in as the submitter and am unable to do anything on Step 5 to re-submit.

https://github.com/user-attachments/assets/9e9588fb-2ae2-4778-aeeb-c65f04165fcc

**Comment by npage123** [comment]
@josbell Found a bug, I think, when saving a draft. I had a req number and date entered when I last saved. Then went back in and deleted the req number and saved again. When returning to review, the req number I removed was still pre-populated. CC: @kimschulke @andressacurvelo 

https://github.com/user-attachments/assets/1cb0bf12-e202-4954-a00d-622296da0ee3

**Comment by npage123** [comment]
@josbell In talking it over with @kimschulke , we came to the conclusion that the the validation of needing at least one of either the req number or date for saving a draft is not needed so if we remove the validations, this bug would then not exist anymore. CC: @andressacurvelo 

<img width="3244" height="1080" alt="Image" src="https://github.com/user-attachments/assets/97c4fa9d-6f11-4506-8f4c-f70c586bda23" />

**Comment by npage123** [comment]
Reviewed with @josbell and we made the decision to document and apply the effort to fix the finding above in a new issue [#6028](https://github.com/HHS/OPRE-OPS/issues/6028).
